### PR TITLE
refactor: standardize fb_action user IO

### PR DESCRIPTION
**Pull request by jamestrew**
Using `vim.notify` for output messages (log level = WARN for areas indicating missteps by users), `vim.ui.input` for inputs.

All messages prefixed by `[telescope]` to be clear about the origins of the message.

Checks off an item from #3

### CODE REVIEW

File: lua/telescope/_extensions/file_browser/utils.lua [hunk 1]
```diff
@@ -147,4 +147,46 @@ fb_utils.group_by_type = function(tbl)
...
 end
 
+local function add_tele_prefix(msg)
+  if not msg:match "^%[telescope%]" then
+    msg = "[telescope] " .. msg
+  end
+  return msg
+end
+
+-- Move this and above functions into telescope core?
```
Comment (tami5) [inline lua/telescope/_extensions/file_browser/utils.lua:157]: Makes sense

**Review by tami5 — COMMENTED** [review]
awesome refactor

**Comment by fdschmidt93** [comment]
Sorry, a bit too lazy right now as I've had worked on tons on the file browser in the weeks before. I'll try to get to have a first brief crack at this either tomorrow or Sunday :)

File: lua/telescope/_extensions/file_browser/utils.lua [hunk 2]
```diff
@@ -147,4 +147,46 @@ fb_utils.group_by_type = function(tbl)
...
+fb_utils.tele_notify = function(msg, log_level, opts)
```
Comment (Conni2461) [inline lua/telescope/_extensions/file_browser/utils.lua:158]: I dont get what you try to achieve with both of these functions. Any styling should be handled by vim.notify, thats the whole point.

For example i saw this:
```
fb_utils.tele_notify "\nRemoving files aborted!"
```

Which results in this output. If you are using https://github.com/rcarriga/nvim-notify

image

the max wrapper around vim.notify i would accept in telescope core is this

```lua
utils.notify = function(msg, log_level, opts)
  vim.notify("[telescope]" .. msg, log_level, opts)
end
```

And then we could replace all the print 
```lua
print "[telescope] Nothing currently selected"
```
with 
```lua
utils.notify "Nothing currently selected"
```
in core

Comment (Conni2461) [reply lua/telescope/_extensions/file_browser/utils.lua:158]: Or we can use the title possibility of `nvim-notify` if its available. So if not `[telescope] msg` else `vim.notify(msg, log_level, { title = "telescope" })`

Comment (tami5) [reply lua/telescope/_extensions/file_browser/utils.lua:158]: > Or we can use the title possibility of `nvim-notify` if its available. So if not `[telescope] msg` else `vim.notify(msg, log_level, { title = "telescope" })`

😋🎉  bingo

**Comment by jamestrew** [comment]
Throwing this PR into draft mode to do some clean up and find a solution to #87, #91

**Comment by jamestrew** [comment]
closes #91 

I've spent some more time looking for solutions for #87 but I'm able to find something that works consistently.
So far, I'm using `vim.api.nvim_command("norm :esc<CR>")` and it works for all actions expect `remove` (due to things being printed prior to the `[y/N]` input getter).  Also for `get_valid_path` if the user completely deletes the default input, it will also print the `Operation aborted` msg on the same line as the input prompt.

**Comment by fdschmidt93** [comment]
I've now looked a bit more at this as I'll eventually will take this over. Couple of notes:

- `vim.ui.input` is async for most users that eg user `dressing.nvim` or `telescope-ui-select`; hence, we cannot rely returning anything from `vim.ui.input`
- `vim.cmd [[redraw!]]` is probably the most reliant way to avoid `hit-enter` prompt

I'll try to work incrementally and push the PR maybe in a week or two.

Thanks again anyways :)

**Pull request closed without merging**